### PR TITLE
docs: add rds iam auth

### PR DESCRIPTION
**Pull request by michaellzc**
add docs for https://github.com/sourcegraph/sourcegraph/pull/47585

this feature was previously requested by a customer (fun fact: they're considering moving to cloud) https://docs.google.com/spreadsheets/d/1Mo9O-sARMdprPpFC_IZIM7AllDvSUzED-9X1fa5SkvI/edit?disco=AAAArZ57qxQ

it still requires more real-world testing, but it's worth having some minimal docs in case someone else want to give it a try.

## Test plan

N/A, docs only

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* main
  * **PR #47999** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/47999" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/47999?utm_source=stack-comment).

**Review by danieldides — APPROVED** [review]

**Review by unknwon — APPROVED** [review]
Thank you!

**Review by filiphaftek — COMMENTED** [review]
Nice!

File: doc/admin/external_services/postgres.md [hunk 1]
```diff
@@ -144,6 +144,45 @@ Most standard PostgreSQL environment variables may be specified (`PGPORT`, etc).
 
 ----
 
+## Usage with AWS RDS IAM Auth
```
Comment (bobheadxi) [inline doc/admin/external_services/postgres.md:147]: I don't think there's a changelog entry for this yet, let's add one :)

**Pull request merged**